File: server/models/reply.js
const mongoose = require("mongoose")

// MongoDB Schemas
const ReplySchema = new mongoose.Schema({
  username: String,
  content: { type: String, required: true },
  image: {
    url: String,
    size: Number,
    width: Number,
    height: Number,
    thumbnailWidth: Number,
    thumbnailHeight: Number
  },
  created: { type: Date, default: Date.now },
  parentReply: { type: mongoose.Schema.Types.ObjectId, ref: 'Reply', default: null },
  threadID: { type: mongoose.Schema.Types.ObjectId, ref: 'Thread', required: true },
  isOP: { type: Boolean, default: false },
  upvotes: {
    count: {type: Number, default: 0},
    ids: [{type: String}]
  },
  downvotes: {
    count: {type: Number, default: 0},
    ids: [{type: String}]
  },
  posterID: String
});

const Reply = mongoose.model('Reply', ReplySchema);

module.exports = Reply;

File: server/models/reply.go
package models

import (
	"go.mongodb.org/mongo-driver/v2/bson"
)

type Reply struct {
	ID          bson.ObjectID      `bson:"_id,omitempty" json:"_id"`
	Username    string             `bson:"username,omitempty" json:"username,omitempty"`
	Content     string             `bson:"content" json:"content"`
	Image       *Image             `bson:"image,omitempty" json:"image,omitempty"`
	Created     bson.DateTime      `bson:"created" json:"created"`
	ParentReply bson.ObjectID      `bson:"parentReply,omitempty" json:"parentReply,omitempty"`
	ThreadID    bson.ObjectID      `bson:"threadID" json:"threadID"`
	IsOP        bool               `bson:"isOP" json:"isOP"`
	Upvotes     *Votes              `bson:"upvotes" json:"upvotes"`
	Downvotes   *Votes              `bson:"downvotes" json:"downvotes"`
	PosterID    string             `bson:"posterID,omitempty" json:"posterID,omitempty"`
}
